File: 1394 find-lucky-integer-in-an-array/solution.java
class Solution {
    public int findLucky(int[] arr) {
     int[] a=new int[501];
     for(int i:arr) a[i]++;
     for(int i=500;i>0;i--){
        if(a[i]==i) return i;
     }
       return -1;
    }
}

File: 1394 find-lucky-integer-in-an-array/solution.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        vector<int> c(1010,0);
        for(int i:arr) c[i]++;
        int m = -1;
        for(int i:arr) {
            if(i==c[i]) m = max(m,i);
        }
        return m;
    }
};
